### PR TITLE
#549 - Added GlobalPermissions interface without implementation

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #549 
Added `GlobalPermissions` interface without implementation

### CODE REVIEW

File: src/main/java/com/artipie/Settings.java [hunk 1]
```diff
@@ -74,6 +76,13 @@
      */
     CompletionStage<Credentials> credentials();
 
+    /**
+     * Artipie global permissions.
+     *
+     * @return Global permissions if configured, empty - otherwise.
+     */
+    CompletionStage<Optional<GlobalPermissions>> permissions();
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/Settings.java:84]: @olegmoz it seems to me that using `Optional` as a return type always adds some unnecessary complications to the code. As far as I understand the idea, absence of global permissions means that we simply use repo permissions as we do now. May be for this case we can simply create and use some special implementation (which simply returns permissions for repository) of `GlobalPermissions` and avoid returning `Optional` here?

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz one suggestion, please, take a look

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz one suggestion, take a look please

File: src/main/java/com/artipie/Settings.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/Settings.java:84]: @olenagerasimova I agree that it adds additional complexity, but we need special handling of `GlobalPermissions` absence. See the table in comments of https://github.com/artipie/artipie/pull/549

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#564` is not assigned, can't get performer

**Comment by 0crat** [comment]
Pull request [#564](https://github.com/artipie/artipie/issues/564) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**